src/YieldDisburser.sol

```diff
@@ -155,9 +158,13 @@ contract YieldDisburser is OwnableUpgradeable {
                 uint256 vote = voterPower * voterDistribution[j];
                 projectDistributions[j] += vote;
                 totalVotes += vote;
+                projectVotes[j] += 1;
             }
             delete holderToDistribution[voter];
         }
+        for (uint256 i; i < projectCount; ++i) {
```

Comment:
Emitting an event in a "get" function is not ideal because it's not technically actually taking any action. Would be best to return data from it and emit the event in the calling function that is actually making storage changes.

Reply:
This function is state changing, as it resets the voting for bread holders. Maybe the function should be renamed?

Reply:
Maybe I'm missing something, but this function does not do that. The calling function `distributeYield()` does that.

Reply:
@bagelface On line 163 the vote is deleted from storage after being counted but we might not need to emit an event here anyway (see my other comment)

Reply:
hmm yea ok yea I misread the function. thought that was a memory variable defined earlier in the function. to be honest, i would like to discuss a larger refactor of this contract with you all because to me it's sort of overly complicated